services/drupal/drupal_service.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+"""Drupal CMS honeypot service, for catching CVE-2018-7600 (Drupalgeddon 2)."""
+from __future__ import unicode_literals
+
+from base_service import ServerCustomService
+
+from drupal_server import DrupalServer, ALERTS, WEB_PORT
+
+EVENT_TYPE_FIELD_NAME = "event_type"
+ORIGINATING_IP_FIELD_NAME = "originating_ip"
+ORIGINATING_PORT_FIELD_NAME = "originating_port"
+REQUEST_FIELD_NAME = "request"
+EXPLOIT_USING_POST = "user/register?element_parents=account/mail/%23value&ajax_form=1&_wrapper_format=drupal_ajax"
+EXPLOIT_USING_GET = "user/password?name[%23post_render][]=passthru&name[%23markup]=id&name[%23type]=markup"
+
+
+class DrupalService(ServerCustomService):
+    """Plugin class for Honeycomb/Mazerunner that runs a Drupal CMS service."""
+
+    def __init__(self, *args, **kwargs):
+        super(DrupalService, self).__init__(*args, **kwargs)
+        self.honeypot = None
+
+    def alert(self, event_name, orig_ip, orig_port, request):
+        """Report an alert to the framework."""
+        params = {
+            EVENT_TYPE_FIELD_NAME: event_name,
+            ORIGINATING_IP_FIELD_NAME: orig_ip,
+            ORIGINATING_PORT_FIELD_NAME: orig_port,
+            REQUEST_FIELD_NAME: request
+        }
+
+        self.add_alert_to_queue(params)
+
+    def on_server_start(self):
+        """Set up a drupal honeypot server."""
+        self.logger.info("{name} received start".format(name=str(self)))
```

Comment:
instead of using str() in the format you can replace with `self.logger.info("{:s} received start".format(self))`